feat: search frappecloud marketplace apps from awesomebar

rutwikhdev opened the pull request:
![search](https://github.com/frappe/frappe/assets/50401596/af9cf1a6-8476-4224-905d-9c0e322a4ba7)

> no-docs

closes https://github.com/frappe/frappe/issues/22174

rutwikhdev committed:
chore: remove unused import

rutwikhdev committed:
Merge branch 'frappe:develop' into develop

rutwikhdev committed:
feat: search frappecloud marketplace apps from awesomebar

ankush commented on frappe/boot.py:
```diff
@@ -442,6 +443,20 @@ def load_currency_docs(bootinfo):
+		remote_site = frappe.conf.frappecloud_url or "frappecloud.com"
+		request_url = f"https://{remote_site}/api/method/marketplace-apps"
+		request = requests.get(request_url, data={"installed_apps": frappe.get_installed_apps()})
+		apps = request.json()['message']
```
cache the output with `shared=True` so only once it will be called per bench.

ankush commented:
We should rank these somewhat lower btw :laughing: `Install POS awesome` shouldn't show up higher than `POS`. Just suppressing ranking manually should be enough for now.

rutwikhdev committed:
chore: cache results and add tracking url

rutwikhdev commented:
> We should rank these somewhat lower btw 😆 `Install POS awesome` shouldn't show up higher than `POS`. Just suppressing ranking manually should be enough for now.

Where do I set this? 🙄

ankush committed:
refactor: get_marketplace_apps

- use generator for setting cache
- drop expiry. No need to keep this updated as of now. Site updates and
  ram usage anyways drops cache.
- Timeout if not able to respond
- Rank results lower than everything else

ankush commented on frappe/boot.py:
```diff
@@ -442,6 +443,28 @@ def load_currency_docs(bootinfo):
+	def get_apps_from_fc():
+		remote_site = frappe.conf.frappecloud_url or "frappecloud.com"
+		request_url = f"https://{remote_site}/api/method/marketplace-apps"
+		request = requests.get(request_url, timeout=5.0)
```
If this endpoint goes down, we could end up DDOSing ourselves :woozy_face: Can't think of any good alternative right now though, we'll rate limit it ig? 

Though this only happens once per bench, but in the event of cache miss + FC not responding it can be problematic.

rutwikhdev committed:
chore: update request url

ankush committed:
fix: avoid excessive retries
